### PR TITLE
cherrypick of variants pr

### PR DESCRIPTION
**Pull request by dzearing**
Cherrypicking variants into 7.0 branch.

### CODE REVIEW

File: packages/theme/src/types/Theme.ts [hunk 1]
```diff
@@ -104,4 +107,5 @@ export interface PartialTheme extends IPartialTheme {
   components?: ComponentStyles;
   tokens?: RecursivePartial<Tokens>;
   stylesheets?: string[];
+  variants?: RecursivePartial<Variants>;
```
Comment (xugao) [inline packages/theme/src/types/Theme.ts:110]: remove :)

**Review by xugao — APPROVED** [review]

File: packages/theme/src/types/Theme.ts (hunk 1 above)
Comment (dzearing) [reply packages/theme/src/types/Theme.ts:110]: you got it!

File: apps/vr-tests/src/stories/ThemeProvider.stories.tsx [hunk 2]
```diff
@@ -54,12 +54,14 @@ storiesOf('ThemeProvider', module)
...
-        tokens: {
-          button: {
-            background: 'yellow',
+        variants: {
+          Button: {
+            root: {
+              background: 'yellow',
+            },
           },
         },
```
Comment (xugao) [inline apps/vr-tests/src/stories/ThemeProvider.stories.tsx:66]:
```suggestion
        components: {
          Button: {
            variants: {
               root: {
                 background: 'yellow',
               },
            }
          },
        },
```

**Pull request merged**